src/components/Reports/GoalCalculator/SharedComponents/GoalCalculatorGrid/GoalCalculatorGrid.test.tsx

```diff
@@ -0,0 +1,232 @@
+import React from 'react';
+import { render, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { SnackbarProvider } from 'notistack';
+import { GoalCalculatorProvider } from '../../Shared/GoalCalculatorContext';
+import { GoalCalculatorGrid } from './GoalCalculatorGrid';
+
+jest.mock('src/lib/intlFormat', () => ({
+  currencyFormat: (value: number, currency: string, locale: string) =>
+    new Intl.NumberFormat(locale, { style: 'currency', currency }).format(
+      value,
+    ),
+}));
+
+const TestWrapper: React.FC<{ children: React.ReactNode }> = ({ children }) => (
+  <SnackbarProvider>
+    <GoalCalculatorProvider>{children}</GoalCalculatorProvider>
+  </SnackbarProvider>
+);
+
+const defaultProps = {
+  categoryName: 'Special Income Name',
+  promptText: 'Add your special income sources',
+  formData: [
+    { id: 1, name: 'Freelance Work', amount: 2500 },
+    { id: 2, name: 'Investment Returns', amount: 1200 },
+    { id: 3, name: 'Rental Income', amount: 1800 },
+  ],
+};
+
+describe('GoalCalculatorGrid', () => {
+  it('renders with initial data and calculates total correctly', async () => {
+    const { getByText } = render(
+      <TestWrapper>
+        <GoalCalculatorGrid {...defaultProps} />
+      </TestWrapper>,
+    );
+
+    expect(getByText('Freelance Work')).toBeInTheDocument();
+    expect(getByText('Investment Returns')).toBeInTheDocument();
+    expect(getByText('Rental Income')).toBeInTheDocument();
+    await waitFor(() => {
+      expect(getByText('Total')).toBeInTheDocument();
+      expect(getByText('$5,500.00')).toBeInTheDocument();
+    });
+  });
+
+  it('adds a new row when Add button is clicked', async () => {
+    const { getByRole, getByText } = render(
+      <TestWrapper>
+        <GoalCalculatorGrid {...defaultProps} />
+      </TestWrapper>,
+    );
+
+    userEvent.click(
+      getByRole('button', {
+        name: /add special income name/i,
+      }),
+    );
+
+    expect(getByText('New Income')).toBeInTheDocument();
+    expect(getByText('$5,500.00')).toBeInTheDocument();
+  });
+
+  it('removes a row when delete button is clicked', async () => {
+    const { getByText, queryByText } = render(
+      <TestWrapper>
+        <GoalCalculatorGrid {...defaultProps} />
+      </TestWrapper>,
+    );
+    const freelanceRow = getByText('Freelance Work').closest('[role="row"]');
+    userEvent.hover(freelanceRow!);
+    const deleteButtons = freelanceRow?.querySelectorAll(
+      '[aria-label="Delete"]',
+    );
+    if (deleteButtons && deleteButtons.length > 0) {
+      userEvent.click(deleteButtons[0] as Element);
+    }
```

Comment:
You should be able to just find the first button with `querySelector`.

```suggestion
    const deleteButton = freelanceRow?.querySelector(
      '[aria-label="Delete"]',
    );
    if (deleteButton) {
      userEvent.click(deleteButton as Element);
    }
```